### PR TITLE
docs: add comparison to similar distributions

### PR DESCRIPTION
**Pull request by andrewrynhard**
Signed-off-by: Andrew Rynhard <andrew@andrewrynhard.com>

### CODE REVIEW

**Comment by andrewrynhard** [comment]
This a first crack at answering a comment question we had on Reddit and HackerNews. I'm not sure I have summed it up well. My of the cuff answers can be seen here:

https://www.reddit.com/r/kubernetes/comments/aqt0u9/talos_a_modern_linux_distribution_for_kubernetes/egj7lqx/

https://news.ycombinator.com/item?id=19175518

File: README.md [hunk 1]
```diff
@@ -137,6 +137,14 @@ See [CHANGELOG.md](CHANGELOG.md)
 > However, the API is based on the Principle of Least Privilege, and exposes only a limited set of methods.
 > We aren't quite there yet, but we envision Talos being a great place for the application of [control theory](https://en.wikipedia.org/wiki/Control_theory) in order to provide a self-healing platform.
 
+**How is Talos different than CoreOS/RancherOS/Linuxkit?**
+
+> Talos does a few things differently.
+> First, Talos is not meant to be a container based Linux distribution for general use.
+> It is targeted for Kubernetes and cannot be used for anything other than that.
+> Another differentiator is that Talos does not provide host-level access.
+> Interaction with the OS is done through an API.
```
Comment (bradbeam) [inline README.md]: Talos is integrated with Kubernetes and cannot be deployed without it. It is not meant to be a general use Linux distribution. 
Talos does not provide host level access ( no shell, ssh, etc ). Interaction with the OS is done through the API via `osctl`.

Thoughts?

Comment (andrewrynhard) [reply README.md]: Much better :)

**Comment by andrewrynhard** [comment]
@bradbeam PTAL. I think this iteration is much more friendly and not so robotic :)

File: README.md [hunk 2]
```diff
@@ -137,6 +137,18 @@ See [CHANGELOG.md](CHANGELOG.md)
...
+
+> Talos is similar in many ways, but there are some differences that make it unique.
+> In some ways you can imagine Talos as a container image, in that it is built with a single purpose in mind.
+> In this case, that purpose is Kubernetes.
+> Talos tightly integrates with Kubernetes, and is not meant to be a general use Linux distribution.
+> This allows us to dramatically decrease the footprint of Talos, and in turn improve a number of other areas like security, and reliability.
+> In addition to this, interaction with the host is done through a gRPC API.
+> You might be wondering how you might run application X if you don't have access to the host.
+> Our answer would be Kubernetes!
+> Simply put, Talos makes a machine a Kubernetes node.
```
Comment (andrewrynhard) [inline README.md]: Maybe "simply put" makes it seem like the previous sentence is what I'm summing up.

File: README.md [hunk 3]
```diff
@@ -137,6 +137,18 @@ See [CHANGELOG.md](CHANGELOG.md)
...
 
+**How is Talos different than CoreOS/RancherOS/Linuxkit?**
+
+> Talos is similar in many ways, but there are some differences that make it unique.
+> In some ways you can imagine Talos as a container image, in that it is built with a single purpose in mind.
+> In this case, that purpose is Kubernetes.
+> Talos tightly integrates with Kubernetes, and is not meant to be a general use Linux distribution.
+> This allows us to dramatically decrease the footprint of Talos, and in turn improve a number of other areas like security, and reliability.
+> In addition to this, interaction with the host is done through a gRPC API.
+> You might be wondering how you might run application X if you don't have access to the host.
```
Comment (andrewrynhard) [inline README.md]: I'm trying to address what one user asked on Reddit: https://www.reddit.com/r/golang/comments/aqx940/talos_a_modern_linux_distribution_for_kubernetes/egjen6i/

**Pull request merged**